query/src/servers/mysql/writers/query_result_writer.rs

```diff
@@ -40,12 +40,14 @@ impl<'a, W: std::io::Write> DFQueryResultWriter<'a, W> {
 
     pub fn write(
         &mut self,
-        query_result: Result<(Vec<DataBlock>, String)>,
+        query_result: Result<(Vec<DataBlock>, String, bool, DataSchemaRef)>,
```

Comment:
It's too complicated to use tuple for this, maybe we should introduce a struct type here?